packages/jest-matchers/src/matchers.js

```diff
@@ -67,6 +67,18 @@ const matchers: MatchersObject = {
           `Received:\n` +
           `  ${printReceived(received)}`
       : () => {
+          if (
+            getType(received) === getType(expected) &&
+            (getType(received) === 'object' || getType(expected) === 'array') &&
+            equals(received, expected, [iterableEquality])
+          ) {
+            return (
+              matcherHint('.toBe') +
+              '\n\n' +
+              'Looks like you wanted to test for object/array equity with strict `toBe` matcher. You probably need to use `toEqual` instead.'
```

Comment:
equity? Shouldn't it be "equality"?

I think it's still useful to keep some of the information from before, at least "Expected value to be (using ===) but they are different objects/arrays. […]”. What do you think?